gson/src/main/java/com/google/gson/internal/bind/ReflectiveTypeAdapterFactory.java

```diff
@@ -417,16 +416,24 @@ T finalize(T accumulator) {
       // we create an Object[] where all primitives are initialized to non-null values.
       constructorArgsDefaults = new Object[parameterTypes.length];
       for (int i = 0; i < parameterTypes.length; i++) {
-        if (parameterTypes[i].isPrimitive()) {
-          // Voodoo magic, we create a new instance of this primitive type using reflection via an
-          // array. The array has 1 element, that of course will be initialized to the primitives
-          // default value. We then retrieve this value back from the array to get the properly
-          // initialized default value for the primitve type.
-          constructorArgsDefaults[i] = Array.get(Array.newInstance(parameterTypes[i], 1), 0);
-        }
+        constructorArgsDefaults[i] = ZEROES.get(parameterTypes[i]);
+        // This will correctly be null for non-primitive types.
```

Comment:
Shouldn't this comment be _above_ the line it is describing?

Reply:
I always hesitate between the two. :-) It feels a bit odd to talk about "this" when it is something we haven't seen yet. But with a trailing colon I think it is OK. Done.